### PR TITLE
Add Block of Coal -> Carbon recipe to the Carbon Press and Compressor

### PR DESCRIPTION
**Pull request by martinbrom**
## Description

Compressor and Carbon Press now accept 8 coal blocks and use those to craft 9 carbon. This is useful when moving and processing large amounts of coal.
Carbon Press coal block recipe gives no processing speed or product amount advantage over using the standard coal one. (9x slower, 9 carbon).

## Changes

- Added recipe to the Compressor multiblock
- Added recipe to the CarbonPress machine

## Related Issues

none, Discord suggestion 915

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [ ] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/CarbonPress.java
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/CarbonPress.java:27]:
```suggestion
        registerRecipe(180, new ItemStack[] { new ItemStack(Material.COAL_BLOCK, 8) }, new ItemStack[] { new SlimefunItemStack(SlimefunItems.CARBON, 9) });
```

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]
Looks good, thank you.
Just one small error I noticed.

**Comment by martinbrom** [comment]
Sorry but I amended the changes out of habit instead of pushing another commit
PS: should the next line where the recipe for compressed carbon is defined also use 'SlimefunItemStack' instead of 'CustomItem' then?

**Review by TheBusyBiscuit — APPROVED** [review]
LGTM

> PS: should the next line where the recipe for compressed carbon is defined also use 'SlimefunItemStack' instead of 'CustomItem' then?

Ideally, yes. Feel free to change that too if you want, otherwise we will change it some day eventually anyway. We generally use our own `ItemStack` implementation nowadays to automatically read and apply CustomModelData and such.

**Comment by martinbrom** [comment]
> Feel free to change that too if you want

I will add it at some point with a few other cleanup things I found while browsing some of the classes

**Comment by TheBusyBiscuit** [comment]
@Slimefun/code-reviewers Guys, this still needs a second review... :eyes:

**Review by WalshyDev — APPROVED** [review]
I assumed we were waiting on @martinbrom per his last message, if not though LGTM

**Comment by TheBusyBiscuit** [comment]
I understood that as in "in a future PR".
Rightfully so though, since this isn't really in scope of this feature anyway.
Anyway, thanks for the PR. I am gonna merge it then!

**Pull request merged**
